Breeding-Insight/bi-web
# [BI-1494] Pedigree Viewer Improvements

# Description
**Story:** [BI-1494 - Improve Pedigree View Display](https://breedinginsight.atlassian.net/browse/BI-1494)

BEFORE MERGE: 
- Merge in https://github.com/Breeding-Insight/BrAPI-Pedigree-Viewer/pull/3, add a new tag for it (see other PR for details), and change the reference in biweb to point to the new tag. 

Work done: 
- Added the option for multiline text in the node, but still supports single line text. 
  - Adjusting all of the spacing to account for the expanded node size. 
- Refactored the pedigree viewer code so it is easier to work with. 
- Added an option to customize font size. 
- Exposed treeLevelPadding and nodeIconGap parameters in the additionalOptions set in the PedigreeViewer call in GermplasmPedigreesView.vue in order to enable possible fiddling of vertical height of links between nodes due to the increase of node heigh to accommodate multiline text. As is the pedigree viewer on BI looks different than that of BreedBase, but is still readable, so parameter adjustment will likely be shaped by stakeholder feedback.. 
- Added a redirect on the germplasm details page for gid paths. Now if you go to `/program/<program id>/germplasm/gid-1`, it will look up the germplasm by accessionNumber and redirect the user to the proper path using the germplasmDbId. 
  - I did it this way because modifying the pedigree viewer to accept a callback to do this would have been a lot of work. 
  - In addition, it is our app that is inconsistent in the use of germplasmDbId and did not want to add a hackaround into the pedigree viewer. This way all of the changes done in this card were general and relevant to the library. 
  - Our app is inconsistent in the use of germplasmDbId because for the germplasm details endpoint we use the BI UUID as a path param, but with all of our germplasm bodies we have the breedbase id as the germplasmDbId. 

# Dependencies
[BrAPI-Pedigree-Viewer: feature/BI-1494](https://github.com/Breeding-Insight/BrAPI-Pedigree-Viewer/pull/3)

# Testing
**Local Setup**
In terminal
- npm install --save ../BrAPI-Pedigree-Viewer
- In the BrAPI-Pedigree-Viewer folder run npm install
- In the BrAPI-Pedigree-Viewer folder run npm run prepare
- In the bi-web folder run npm install 

In IntelliJ
- In bi-web add file .eslintignore with line ../BrAPI-Pedigree-Viewer/build
- Restart bi-web

**Test Steps**
- Import some germplasm with pedigrees
- Look at the details of one of them 
- Look at pedigree nodes and ensure they are in format of <Name> [GID: <GID>]
- Do lots of expansions on pedigree nodes
- Try to navigating to some other germplasm by clicking the pedigree viewer nodes, ensure redirect is to correct germplasm 

# Checklist:

- [ ] I have run TAF: [_\<link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/2671696187)
